modules/zeromq/src/test/java/org/apache/ignite/stream/zeromq/IgniteZeroMqStreamerTest.java

```diff
@@ -0,0 +1,233 @@
+ */
+public class IgniteZeroMqStreamerTest extends GridCommonAbstractTest {
+    /** Cache entries count. */
+    private static final int CACHE_ENTRY_COUNT = 1000;
+
+    /** Local address for 0mq. */
+    private final String ADDR = "tcp://localhost:5671";
+
+    /** Topic name for PUB-SUB. */
+    private final byte[] TOPIC = "0mq".getBytes();
+
+    /** Constructor. */
+    public IgniteZeroMqStreamerTest() {
+        super(true);
+    }
+
+    /** {@inheritDoc} */
+    @Override protected long getTestTimeout() {
+        return 10_000;
+    }
+
+    /** {@inheritDoc} */
+    @Override public void beforeTestsStarted() throws Exception {
+        grid().getOrCreateCache(defaultCacheConfiguration());
+    }
+
+    /** {@inheritDoc} */
+    @Override public void afterTestsStopped() throws Exception {
+        stopAllGrids();
+    }
+
+    /**
+     * @throws Exception Test exception.
+     */
+    public void testZeroMqPairSocket() throws Exception {
+        try (IgniteDataStreamer<Integer, String> dataStreamer = grid().dataStreamer(null)) {
+            try (IgniteZeroMqStreamer streamer = newStreamerInstance(
+                dataStreamer, 1, ZeroMqTypeSocket.PAIR, ADDR, null);) {
+                executeStreamer(streamer, ZMQ.PAIR, null);
+            }
+        }
+    }
+
+    /**
+     * @throws Exception Test exception.
+     */
+    public void testZeroMqSubSocket() throws Exception {
+        try (IgniteDataStreamer<Integer, String> dataStreamer = grid().dataStreamer(null)) {
+            try (IgniteZeroMqStreamer streamer = newStreamerInstance(
+                dataStreamer, 3, ZeroMqTypeSocket.SUB, ADDR, TOPIC);) {
+                executeStreamer(streamer, ZMQ.PUB, TOPIC);
+            }
+        }
+    }
+
+    /**
+     * @throws Exception Test exception.
+     */
+    public void testZeroMqPullSocket() throws Exception {
+        try (IgniteDataStreamer<Integer, String> dataStreamer = grid().dataStreamer(null)) {
+            try (IgniteZeroMqStreamer streamer = newStreamerInstance(
+                dataStreamer, 4, ZeroMqTypeSocket.PULL, ADDR, null);) {
+                executeStreamer(streamer, ZMQ.PUSH, null);
+            }
+        }
+    }
+
+    /**
+     * Execute ZeroMQ streamer and checking cache content after streaming finished.
+     * Set singleTupleExtractor via {@link ZeroMqStringSingleTupleExtractor} in streamer.
+     *
+     * @param streamer ZeroMQ streamer.
+     * @param clientSocket .
+     * @param topic .
+     * @throws InterruptedException Test exception.
+     */
+    private void executeStreamer(IgniteZeroMqStreamer streamer, int clientSocket,
+        byte[] topic) throws InterruptedException {
+        streamer.setSingleTupleExtractor(new ZeroMqStringSingleTupleExtractor());
+
+        IgniteCache<Integer, String> cache = grid().cache(null);
+
+        CacheListener listener = subscribeToPutEvents();
+
+        assertEquals(0, cache.size(CachePeekMode.PRIMARY));
```

Comment:
please move it after `streamer.start()` and before `startZeroMqClient(clientSocket, topic)`